modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriter.java

```diff
@@ -108,7 +113,12 @@ public void writeEntityBody(Task item, JsonGenerator jg) throws IOException {
             if (StringUtils.isNotBlank(workflowInstanceId)) {
                 NodeAccessRunner nodeAccessRunner = new NodeAccessRunner(wrapper.getSession(), workflowInstanceId,
                         nodeId);
-                nodeAccessRunner.runUnrestricted();
+                try {
+                    nodeAccessRunner.runUnrestricted();
+                } catch (DocumentNotFoundException e) {
+                    // If the DocumentRoute is not found (deleted for example), the TaskWriter fails.
```

Comment:
The comment is not accurate, the TaskWriter does not fail anymore with your fix. Keep it simple...
```
// Workflow instance not found
```
I'm not sure it's even needed according to the `DocumentNotFoundException` and the log message?

Reply:
At the begining the catch was at the end of the try resource (way more line below) so I explained why I was catching.
I should have removed it when I made a very local try catch